src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,173 @@
 import fr.esgi.al5_2.Tayarim.auth.JwtHelper;
 import fr.esgi.al5_2.Tayarim.auth.TokenCacheService;
 import fr.esgi.al5_2.Tayarim.dto.auth.AuthLoginResponseDTO;
+import fr.esgi.al5_2.Tayarim.dto.proprietaire.AdministrateurDTO;
 import fr.esgi.al5_2.Tayarim.dto.proprietaire.ProprietaireDTO;
-import fr.esgi.al5_2.Tayarim.entities.Proprietaire;
-import fr.esgi.al5_2.Tayarim.exceptions.*;
+import fr.esgi.al5_2.Tayarim.exceptions.AdministrateurNotFoundException;
+import fr.esgi.al5_2.Tayarim.exceptions.ProprietaireNotFoundException;
+import fr.esgi.al5_2.Tayarim.exceptions.TokenExpireOrInvalidException;
+import fr.esgi.al5_2.Tayarim.exceptions.UtilisateurNotFoundException;
 import lombok.NonNull;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.AbstractMap;
+import java.util.Map.Entry;
 import java.util.UUID;
 
+/**
+ * Classe de service gérant l'authentification des utilisateurs.
+ */
 @Service
 @Transactional(readOnly = true)
 public class AuthService {
 
-    private final ProprietaireService proprietaireService;
-    private final JwtHelper jwtHelper;
+  private final ProprietaireService proprietaireService;
+  private final AdministrateurService administrateurService;
+  private final JwtHelper jwtHelper;
+
+  private final TokenCacheService tokenCacheService;
+
+  /**
+   * Constructeur pour le service d'authentification.
+   *
+   * @param proprietaireService Le service de gestion des propriétaires.
+   * @param administrateurService Le service de gestion des administrateurs.
+   * @param jwtHelper L'outil d'assistance JWT.
+   * @param tokenCacheService Le service de cache des tokens.
+   */
+  public AuthService(ProprietaireService proprietaireService,
+      AdministrateurService administrateurService, JwtHelper jwtHelper,
+      TokenCacheService tokenCacheService) {
+    this.proprietaireService = proprietaireService;
+    this.administrateurService = administrateurService;
+    this.jwtHelper = jwtHelper;
+    this.tokenCacheService = tokenCacheService;
+  }
+
+  /**
+   * Tente de connecter un utilisateur avec son email et mot de passe.
+   *
+   * @param email L'email de l'utilisateur.
+   * @param password Le mot de passe de l'utilisateur.
+   * @return Un objet {@link AuthLoginResponseDTO} contenant les informations de l'utilisateur et un token.
+   * @throws UtilisateurNotFoundException Si aucun utilisateur n'est trouvé avec cet email.
+   * @throws ProprietaireNotFoundException Si le propriétaire est trouvé mais le mot de passe est incorrect.
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 108).